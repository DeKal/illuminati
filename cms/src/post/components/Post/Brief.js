import React from 'react'
import PropTypes from 'prop-types'
import TextField from 'post/components/TextField'

const Brief = ({ isEdited, brief, editPostBrief }) => (
  <TextField
    label="Post Brief"
    required
    variant="outlined"
    value={brief}
    multiline
    rows={6}
    rowsMax={8}
    fullWidth
    error={isEdited}
    onChange={event => editPostBrief(event.target.value)}
  />
)

export default Brief

Brief.propTypes = {
  isEdited: PropTypes.bool,
  brief: PropTypes.string,
  editPostBrief: PropTypes.func
}
