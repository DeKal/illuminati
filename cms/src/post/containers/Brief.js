import { connect } from 'react-redux'
import { getPostBrief, getEditedBrief } from 'post/state/selectors'
import { editPostBrief } from 'post/state/actions'
import Brief from 'post/components/Brief'

const mapStateToProps = state => {
  const editedBrief = getEditedBrief(state)
  const brief = getPostBrief(state)
  return {
    brief: editedBrief || '',
    isEdited: editedBrief !== brief
  }
}

const mapDispatchToProps = {
  editPostBrief
}

export default connect(mapStateToProps, mapDispatchToProps)(Brief)
