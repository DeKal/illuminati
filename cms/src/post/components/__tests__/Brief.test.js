import * as React from 'react'
import { shallow } from 'enzyme'
import Brief from '../Brief'

describe('Brief', () => {
  it('should render Brief correctly', function() {
    const wrap = shallow(<Brief brief="brief" />)
    expect(wrap).toMatchSnapshot()
  })
})
