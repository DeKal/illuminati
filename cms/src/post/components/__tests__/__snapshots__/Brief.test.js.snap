// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brief should render Brief correctly 1`] = `
<WithStyles(WithStyles(ForwardRef(TextField)))
  fullWidth={true}
  label="Post Brief"
  multiline={true}
  onChange={[Function]}
  required={true}
  rows={6}
  rowsMax={8}
  value="brief"
  variant="outlined"
/>
`;
